packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -373,15 +387,23 @@ export function jsxProdSignatureRunningInDevWithDynamicChildren(
 ) {
   if (__DEV__) {
     const isStaticChildren = false;
+    const trackActualOwner =
+      ReactSharedInternals.recentlyCreatedOwnerStacks++ < ownerStackLimit;
```

Comment:
It's weird to me that the owner stack value below is gated by `__DEV__` but this line is not?

Reply:
Right, I assumed it gets DCE'd but since it has a side-effect, GCC couldn't safely do that. So the explicit `__DEV__ && ...` was needed here.